lib/core/tree.js

```diff
@@ -0,0 +1,51 @@
+const { wellknownHeaderNames } = require('./constants')
+
+class Tree {
+  #node
+  constructor () {
+    this.#node = {}
+  }
+
+  /**
+   * @param {string} value
+   */
+  insert (value) {
+    const target = Buffer.from((value = value.toLowerCase()))
+    let node = this.#node
+    for (let i = 0; i < target.length; ++i) {
+      const key = target[i]
+      node[key] ??= {}
+      if (key >= 0x61 && key <= 0x7a) {
+        node[key & ~32] ??= node[key]
+      }
+      node = node[key]
+    }
+    node.value = value
+  }
+
+  /**
+   * @param {Uint8Array} buffer
+   * @returns {string | null}
+   */
+  lookup (buffer) {
+    const length = buffer.length
+    let node = this.#node
+    for (let i = 0; i < length; ++i) {
+      const key = buffer[i]
+      if (!(key in node) || node === undefined) return null
```

Comment:
Why do you need `key in node`?
```suggestion
      if (node === undefined) return null
```

Reply:
In this case, it looks like you don't have to.

Reply:
@tsctx remove it and run benchmark again?

Reply:
```
• parseHeaders
--------------------------------------------------- -----------------------------
default        4.08 µs/iter     (3.3 µs … 905.7 µs)    3.8 µs   10.1 µs   13.2 µs
allowUnsafe    2.38 µs/iter     (2.06 µs … 4.06 µs)   2.62 µs   4.06 µs   4.06 µs

summary for parseHeaders
  allowUnsafe
   1.72x faster than default

• parseHeaders (irregular)
--------------------------------------------------- -----------------------------
default        4.14 µs/iter     (3.88 µs … 5.57 µs)   4.05 µs   5.57 µs   5.57 µs
allowUnsafe    2.24 µs/iter     (2.06 µs … 2.92 µs)   2.25 µs   2.92 µs   2.92 µs

summary for parseHeaders (irregular)
  allowUnsafe
   1.85x faster than default
```

Reply:
It was a little fast.